src/index.ts

```diff
@@ -1,49 +1,127 @@
-import * as array from 'fp-ts/lib/Array';
-import {fold} from 'fp-ts/lib/Either';
-import {map} from 'fp-ts/lib/Option';
-import {pipe} from 'fp-ts/lib/pipeable';
+import * as A from 'fp-ts/lib/Array';
+import * as E from 'fp-ts/lib/Either';
+import * as NEA from 'fp-ts/lib/NonEmptyArray';
+import * as O from 'fp-ts/lib/Option';
+import * as R from 'fp-ts/lib/Record';
+import { pipe } from 'fp-ts/lib/pipeable';
 import * as t from 'io-ts';
 
+import { takeUntil } from './utils';
+
+const isUnionType = ({ type }: t.ContextEntry) => type instanceof t.UnionType;
+
 const jsToString = (value: t.mixed) =>
   value === undefined ? 'undefined' : JSON.stringify(value);
 
-export const formatValidationError = (error: t.ValidationError) => {
```

Comment:
Ouch. This was a breaking change for me :D

Reply:
Ohh sorry! I'll find a way to fix it today.

Now there's a version for union errors and "other" errors. Because errors on union types are basically treated as one error per case of the union type so what I did was group by the path of the property that had an error. I'll look into a way of generalizing that and if it's not possible I might have to export both functions.

Reply:
Check #33 

I also added another exported function that can be used to format the errors 'in place' 

```ts
import * as t from 'io-ts';
import { formatValidationErrors } from 'io-ts-reporters';
import * as E from 'fp-ts/lib/Either';
import { pipe } from 'fp-ts/lib/pipeable';

const User = t.interface({ name: t.string });
pipe({ nam: 'Jane' }, User.decode, E.mapLeft(formatValidationErrors));
```